core/keyboard_nav/ast_node.ts

```diff
@@ -724,25 +716,14 @@ export class ASTNode {
    * @returns An AST node pointing to a connection.
    */
   static createConnectionNode(connection: Connection): ASTNode | null {
-    if (!connection) {
-      return null;
-    }
     const type = connection.type;
-    if (type === ConnectionType.INPUT_VALUE) {
-      // AnyDuringMigration because:  Argument of type 'Input | null' is not
-      // assignable to parameter of type 'Input'.
-      return ASTNode.createInputNode(
-        connection.getParentInput() as AnyDuringMigration,
-      );
-    } else if (
-      type === ConnectionType.NEXT_STATEMENT &&
-      connection.getParentInput()
+    if (
+      type === ConnectionType.INPUT_VALUE ||
+      (type === ConnectionType.NEXT_STATEMENT && connection.getParentInput())
     ) {
-      // AnyDuringMigration because:  Argument of type 'Input | null' is not
-      // assignable to parameter of type 'Input'.
-      return ASTNode.createInputNode(
-        connection.getParentInput() as AnyDuringMigration,
-      );
+      const parentInput = connection.getParentInput();
+      if (!parentInput) return null;
```

Comment:
You shouldn't need to check if the parent input is null here, since you checked it in the condition.

Reply:
Done.